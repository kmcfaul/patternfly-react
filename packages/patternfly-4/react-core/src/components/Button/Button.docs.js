import { Button, ButtonVariant, ButtonType } from '@patternfly/react-core';
import VariantsExample from './examples/ButtonVariants';
import BlockExample from './examples/BlockButton';
import LinkExample from './examples/LinkButton';

export default {
  title: 'Button',
  components: {
    Button
  },
  enumValues: {
    'Object.values(ButtonVariant)': Object.values(ButtonVariant),
    'Object.values(ButtonType)': Object.values(ButtonType)
  },
  examples: [
    { component: VariantsExample, title: 'Button Variants' },
    { component: BlockExample, title: 'Block Button' },
    {
      component: LinkExample,
      title: 'Links',
      description: `Links with button styling. Semantic buttons and links are important for usability as well as accessibility. Using an "a" instead of a "button" element to perform user initiated actions should be avoided, unless absolutely necessary.`
    }
  ]
};
