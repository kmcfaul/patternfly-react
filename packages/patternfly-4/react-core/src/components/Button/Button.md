---
title: 'Button'
cssPrefix: 'pf-c-button'
---

## Block button

import { Button } from '@patternfly/react-core';
import { TimesIcon, PlusCircleIcon } from '@patternfly/react-icons';

```js
import React from 'react';
import { Button } from '@patternfly/react-core';
import { TimesIcon, PlusCircleIcon } from '@patternfly/react-icons';

class BlockButton extends React.Component {
  render() {
    return <Button isBlock>Block level button</Button>;
  }
}
```

## Button variants

```js
import React from 'react';
import { Button, ButtonVariant } from '@patternfly/react-core';
import { TimesIcon, PlusCircleIcon } from '@patternfly/react-icons';

class ButtonVariants extends React.Component {
  render() {
    return (
      <React.Fragment>
        <Button variant={ButtonVariant.primary}>Primary</Button>{' '}
        <Button variant={ButtonVariant.secondary}>Secondary</Button>{' '}
        <Button variant={ButtonVariant.tertiary}>Tertiary</Button>{' '}
        <Button variant={ButtonVariant.danger}>Danger</Button>{' '}
        <Button variant={ButtonVariant.link}>
          <PlusCircleIcon /> Link button
        </Button>{' '}
        <Button variant={ButtonVariant.plain} aria-label="Action">
          <TimesIcon />
        </Button>
        <Button variant={ButtonVariant.link} isInline>
          Inline Link Button
        </Button>
      </React.Fragment>
    );
  }
}
```

## Link button

```js
import React from 'react';
import { Button, ButtonVariant } from '@patternfly/react-core';
import { TimesIcon, PlusCircleIcon } from '@patternfly/react-icons';

class LinkButton extends React.Component {
  render() {
    return (
      <React.Fragment>
        <Button component="a" href="https://pf-next.com/" target="_blank" variant={ButtonVariant.primary}>
          Link to Core Docs
        </Button>{' '}
        <Button component="a" href="https://pf-next.com/" target="_blank" variant={ButtonVariant.secondary}>
          Secondary Link to Core Docs
        </Button>{' '}
        <Button component="a" isDisabled href="https://pf-next.com/" target="_blank" variant={ButtonVariant.tertiary}>
          Tertiary Link to Core Docs
        </Button>
      </React.Fragment>
    );
  }
}
```
