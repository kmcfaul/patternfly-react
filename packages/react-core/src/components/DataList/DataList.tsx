import * as React from 'react';
import { css } from '@patternfly/react-styles';
import styles from '@patternfly/react-styles/css/components/DataList/data-list';
import { PickOptional } from '../../helpers/typeUtils';

export enum DataListWrapModifier {
  nowrap = 'nowrap',
  truncate = 'truncate',
  breakWord = 'breakWord'
}

export interface DataListProps extends React.HTMLProps<HTMLUListElement> {
  /** Content rendered inside the DataList list */
  children?: React.ReactNode;
  /** Additional classes added to the DataList list */
  className?: string;
  /** Adds accessible text to the DataList list */
  'aria-label': string;
  /** Optional callback to make DataList selectable, fired when DataListItem selected */
  onSelectDataListItem?: (id: string) => void;
  /** Optional callback to make DataList draggable, fired when dragging ends */
  onDragFinish?: (newList: React.ReactElement[]) => void;
  /** Optional informational callback for dragging, fired when dragging starts */
  onDragStart?: (id: string) => void;
  /** Optional informational callback for dragging, fired when an item moves */
  onDragMove?: (oldIndex: number, newIndex: number) => void;
  /** Optional informational callback for dragging, fired when dragging is cancelled */
  onDragCancel?: () => void;
  /** Id of DataList item currently selected */
  selectedDataListItemId?: string;
  /** Flag indicating if DataList should have compact styling */
  isCompact?: boolean;
  /** Determines which wrapping modifier to apply to the DataList */
  wrapModifier?: DataListWrapModifier | 'nowrap' | 'truncate' | 'breakWord';
}

interface DataListState {
  childrenCopy: React.ReactNode;
  renderable: React.ReactElement[];
  draggedItem: React.ReactElement;
  dragging: boolean;
  to: number;
}

interface DataListContextProps {
  isSelectable: boolean;
  selectedDataListItemId: string;
  updateSelectedDataListItem: (id: string) => void;
  isDraggable: boolean;
  dragStart: (e: React.DragEvent) => void;
  dragEnd: (e: React.DragEvent) => void;
  dragKeyHandler: (e: React.KeyboardEvent) => void;
}

export const DataListContext = React.createContext<Partial<DataListContextProps>>({
  isSelectable: false
});

export class DataList extends React.Component<DataListProps, DataListState> {
  static displayName = 'DataList';
  static defaultProps: PickOptional<DataListProps> = {
    children: null,
    className: '',
    selectedDataListItemId: '',
    isCompact: false,
    wrapModifier: null
  };
  dragFinished: boolean = false;
  arrayCopy: React.ReactElement[] = React.Children.toArray(this.props.children) as React.ReactElement[];

  state: DataListState = {
    childrenCopy: this.props.children,
    renderable: React.Children.toArray(this.props.children) as React.ReactElement[],
    to: -1,
    draggedItem: null,
    dragging: false
  };

  static getDerivedStateFromProps(props: DataListProps, state: DataListState) {
    if (props.children !== state.childrenCopy) {
      return {
        renderable: React.Children.toArray(props.children) as React.ReactElement[],
        childrenCopy: props.children
      };
    }

    return null;
  }

  componentDidUpdate() {
    if (this.dragFinished) {
      this.arrayCopy = React.Children.toArray(this.props.children) as React.ReactElement[];
      this.dragFinished = false;

      this.setState({
        renderable: React.Children.toArray(this.props.children) as React.ReactElement[],
        to: -1,
        draggedItem: null
      });
    }
  }

  getIndex = (id: string) => this.state.renderable.findIndex(item => item.props.id === id);

  move = (arr: React.ReactElement[], oldIndex: number, newIndex: number) => {
    const { onDragMove } = this.props;
    const { draggedItem } = this.state;
    const ghost = React.cloneElement(draggedItem, {
      className: css(draggedItem.props.className, styles.modifiers.ghostRow),
      'aria-pressed': true
    });

    arr.splice(oldIndex, 1);
    arr.splice(newIndex, 0, ghost);
    onDragMove(oldIndex, newIndex);
    return arr;
  };

  dragStart = (evt: React.DragEvent) => {
    const { children, onDragStart } = this.props;
    evt.dataTransfer.effectAllowed = 'move';
    evt.dataTransfer.setData('text/plain', evt.currentTarget.id);
    // e.dataTransfer.setDragImage(React.createElement('div'), 0, 0);

    this.setState({
      draggedItem: React.Children.toArray(children).find(
        item => (item as React.ReactElement).props.id === evt.currentTarget.id
      ) as React.ReactElement
    });
    onDragStart(evt.currentTarget.id);
  };

  dragEnd = (evt: React.DragEvent) => {
    const { to, renderable, draggedItem } = this.state;
    const { onDragFinish } = this.props;

    const data = renderable;
    data.splice(to, 1, draggedItem);
    this.dragFinished = true;
    onDragFinish(data);
  };

  dragOver = (evt: React.DragEvent) => {
    const { to } = this.state;
    evt.preventDefault();
    const currListItem = (evt.target as Element).closest('li');

    const currInd = currListItem ? this.getIndex(currListItem.id) : 0;
    if (currInd !== to) {
      this.setState(prevState => ({
        to: currInd,
        renderable: this.move(this.arrayCopy, prevState.to === -1 ? currInd : prevState.to, currInd)
      }));
    }
  };

  handleDragButtonKeys = (evt: React.KeyboardEvent) => {
    const { to, draggedItem, dragging, renderable } = this.state;
    const { children, onDragFinish, onDragStart, onDragCancel } = this.props;
    if (
      evt.key !== ' ' &&
      evt.key !== 'Escape' &&
      evt.key !== 'Enter' &&
      evt.key !== 'ArrowUp' &&
      evt.key !== 'ArrowDown'
    ) {
      if (dragging) {
        this.setState({
          dragging: false
        });
        this.dragFinished = true;
      }
      return;
    }
    evt.preventDefault();

    const dragItem = (evt.target as Element).closest('li');
    const dragInd = this.getIndex(dragItem.id);

    if (evt.key === ' ' || (evt.key === 'Enter' && !dragging)) {
      onDragStart(dragItem.id);
      this.setState({
        dragging: true,
        draggedItem: React.Children.toArray(children).find(
          item => (item as React.ReactElement).props.id === dragItem.id
        ) as React.ReactElement
      });
    } else if (dragging) {
      if (evt.key === 'Escape' || evt.key === 'Enter') {
        this.setState({
          dragging: false
        });
        this.dragFinished = true;
        if (evt.key === 'Enter') {
          const data = renderable;
          data.splice(to, 1, draggedItem);
          onDragFinish(data);
        } else {
          onDragCancel();
        }
      } else if (evt.key === 'ArrowUp') {
        this.setState(prevState => {
          const prevTo = prevState.to === -1 ? dragInd : prevState.to;
          const next = prevTo - 1 >= 0 ? prevTo - 1 : 0;
          return {
            to: next,
            renderable: this.move(this.arrayCopy, prevTo, next)
          };
        });
      } else if (evt.key === 'ArrowDown') {
        this.setState(prevState => {
          const prevTo = prevState.to === -1 ? dragInd : prevState.to;
          const next = prevTo + 1 < this.arrayCopy.length ? prevTo + 1 : this.arrayCopy.length - 1;
          return {
            to: next,
            renderable: this.move(this.arrayCopy, prevTo, next)
          };
        });
      }
    }
  };

  render() {
    const {
      className,
      children,
      onSelectDataListItem,
      selectedDataListItemId,
      isCompact,
      onDragStart,
      onDragMove,
      onDragCancel,
      onDragFinish,
      wrapModifier,
      ...props
    } = this.props;
    const { renderable } = this.state;
    const isSelectable = onSelectDataListItem !== undefined;
    const isDraggable = onDragFinish !== undefined;

    const updateSelectedDataListItem = (id: string) => {
      onSelectDataListItem(id);
    };

    const dragProps = isDraggable && {
      onDragOver: this.dragOver,
      onDrop: this.dragOver
    };

    return (
      <DataListContext.Provider
        value={{
          isSelectable,
          selectedDataListItemId,
          updateSelectedDataListItem,
          isDraggable,
          dragStart: this.dragStart,
          dragEnd: this.dragEnd,
          dragKeyHandler: this.handleDragButtonKeys
        }}
      >
        <ul
          className={css(
            styles.dataList,
            isCompact && styles.modifiers.compact,
            wrapModifier && styles.modifiers[wrapModifier],
            className
          )}
          {...props}
          {...dragProps}
        >
          {isDraggable ? renderable : children}
        </ul>
      </DataListContext.Provider>
    );
  }
}
