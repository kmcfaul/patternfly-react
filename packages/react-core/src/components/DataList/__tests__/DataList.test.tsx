import React from 'react';

import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { DataList } from '../DataList';
import { DataListItem } from '../DataListItem';
import { DataListAction } from '../DataListAction';
import { DataListCell } from '../DataListCell';
import { DataListToggle } from '../DataListToggle';
import { DataListItemCells } from '../DataListItemCells';
import { DataListItemRow } from '../DataListItemRow';
import { DataListContent } from '../DataListContent';
import { Button } from '../../Button';

import styles from '@patternfly/react-styles/css/components/DataList/data-list';

test('Renders to match snapshot', () => {
  const { asFragment } = render(<DataList aria-label="list" />);
  expect(screen.getByLabelText('list')).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});

test(`Renders ${styles.modifiers.compact} when isCompact = true`, () => {
  render(<DataList aria-label="list" isCompact />);
  expect(screen.getByLabelText('list')).toHaveClass(styles.modifiers.compact);
});

const gridBreakpointClasses = {
  none: styles.modifiers.gridNone,
  always: 'pf-m-grid',
  sm: styles.modifiers.gridSm,
  md: styles.modifiers.gridMd,
  lg: styles.modifiers.gridLg,
  xl: styles.modifiers.gridXl,
  '2xl': styles.modifiers.grid_2xl
};

['none', 'always', 'sm', 'md', 'lg', 'xl', '2xl'].forEach((oneBreakpoint) => {
  test(`Has breakpoint - ${oneBreakpoint} when gridBreakpoint is pased`, () => {
    render(<DataList aria-label="list" gridBreakpoint={oneBreakpoint as any} />);
    expect(screen.getByLabelText('list')).toHaveClass(gridBreakpointClasses[oneBreakpoint]);
  });
});

test(`Has breakpoint - 2xl when gridBreakpoint is pased`, () => {
  render(<DataList aria-label="list" gridBreakpoint="2xl" />);
  expect(screen.getByLabelText('list')).toHaveClass(styles.modifiers['grid_2xl']);
});

test('Renders custom class when passed', () => {
  render(<DataList key="list-id-1" className="data-list-custom" aria-label="list" />);
  expect(screen.getByLabelText('list')).toHaveClass('data-list-custom');
});

test('List renders with a hidden input to improve a11y when selectableRow is passed', () => {
  render(
    <DataList aria-label="this is a simple list" onSelectableRowChange={() => {}}>
      <DataListItem>
        <DataListItemRow aria-labelledby="test-id">
          <p id="test-id">Test</p>
        </DataListItemRow>
      </DataListItem>
    </DataList>
  );

  const selectableInput = screen.getByRole('radio', { hidden: true });

  expect(selectableInput).toBeInTheDocument();
});

test('List does not render with a hidden input to improve a11y when selectableRow is not passed', () => {
  render(
    <DataList aria-label="this is a simple list">
      <DataListItem>
        <DataListItemRow aria-labelledby="test-id">
          <p id="test-id">Test</p>
        </DataListItemRow>
      </DataListItem>
    </DataList>
  );

  const selectableInput = screen.queryByRole('radio', { hidden: true });

  expect(selectableInput).not.toBeInTheDocument();
});

test('List calls selectableRow.onChange when the selectable input changes', async () => {
  const mock = jest.fn();
  const user = userEvent.setup();

  render(
    <DataList aria-label="this is a simple list" onSelectableRowChange={mock} selectedDataListItemId="">
      <DataListItem id="item-test-id">
        <DataListItemRow aria-labelledby="test-id">
          <p id="test-id">Test</p>
        </DataListItemRow>
      </DataListItem>
    </DataList>
  );

  const selectableInput = screen.getByRole('radio', { hidden: true });
  await user.click(selectableInput);

  expect(mock).toHaveBeenCalled();
});

test('List does not call selectableRow.onChange when the selectable input is not changed', () => {
  const mock = jest.fn();

  render(
    <DataList aria-label="this is a simple list" onSelectableRowChange={mock} selectedDataListItemId="">
      <DataListItem id="item-test-id">
        <DataListItemRow aria-labelledby="test-id">
          <p id="test-id">Test</p>
        </DataListItemRow>
      </DataListItem>
    </DataList>
  );

  expect(mock).not.toHaveBeenCalled();
});

test('Item applies selectableInputAriaLabel to the hidden input', () => {
  render(
    <DataList aria-label="this is a simple list" onSelectableRowChange={() => {}}>
      <DataListItem selectableInputAriaLabel="Data list item label test">
        <DataListItemRow aria-labelledby="test-id">
          <p id="test-id">Test</p>
        </DataListItemRow>
      </DataListItem>
    </DataList>
  );

  const selectableInput = screen.getByRole('radio', { hidden: true });

  expect(selectableInput).toHaveAccessibleName('Data list item label test');
});

test('Item defaults to labelling its input using its aria-labelledby prop', () => {
  render(
    <DataList aria-label="this is a simple list" onSelectableRowChange={() => {}}>
      <DataListItem aria-labelledby="test-id">
        <p id="test-id">Test cell content</p>
      </DataListItem>
    </DataList>
  );

  const selectableInput = screen.getByRole('radio', { hidden: true });

  expect(selectableInput).toHaveAccessibleName('Test cell content');
});

test('Item prioritizes selectableInputAriaLabel over aria-labelledby prop', () => {
  render(
    <DataList aria-label="this is a simple list" onSelectableRowChange={() => {}}>
      <DataListItem aria-labelledby="test-id" selectableInputAriaLabel="Data list item label test">
        <p id="test-id">Test cell content</p>
      </DataListItem>
    </DataList>
  );

  const selectableInput = screen.getByRole('radio', { hidden: true });

  expect(selectableInput).toHaveAccessibleName('Data list item label test');
});

test('Item renders to match snapshot', () => {
  const { asFragment } = render(
    <DataListItem key="item-id-1" aria-labelledby="item-1">
      test
    </DataListItem>
  );
  expect(asFragment()).toMatchSnapshot();
});

test(`Item has ${styles.modifiers.expanded} when isExpanded is passed`, () => {
  render(
    <DataListItem aria-labelledby="item-1" isExpanded>
      test
    </DataListItem>
  );
  expect(screen.getByRole('listitem')).toHaveClass(styles.modifiers.expanded);
});

test('Item renders with custom class name', () => {
  render(
    <DataListItem className="data-list-item-custom" aria-labelledby="item-1">
      test
    </DataListItem>
  );
  expect(screen.getByRole('listitem')).toHaveClass('data-list-item-custom');
});

test('Item row renders to match snapshot', () => {
  const { asFragment } = render(<DataListItemRow>test</DataListItemRow>);
  expect(asFragment()).toMatchSnapshot();
});

test('Cell renders to match snapshot', () => {
  const { asFragment } = render(<DataListCell>Secondary</DataListCell>);
  expect(asFragment()).toMatchSnapshot();
});

test('Cells renders to match snapshot', () => {
  const { asFragment } = render(
    <DataListItemCells
      dataListCells={[
        <DataListCell key="list-id-1" id="primary-item" className="data-list-custom">
          Primary Id
        </DataListCell>,
        <DataListCell key="list-id-2" id="primary-item" className="data-list-custom">
          Primary Id 2
        </DataListCell>
      ]}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});

test('Cell has width modifier when width is passed', () => {
  [
    { width: 1, class: '' },
    { width: 2, class: 'pf-m-flex-2' },
    { width: 3, class: 'pf-m-flex-3' },
    { width: 4, class: 'pf-m-flex-4' },
    { width: 5, class: 'pf-m-flex-5' }
  ].forEach((testCase, index) => {
    const testId = `cell-test-id-${index}`;

    render(
      <DataListCell data-testid={testId} width={testCase.width as any} key={index}>
        Primary Id
      </DataListCell>
    );

    const dataListCell = screen.getByTestId(testId);

    testCase.class === ''
      ? expect(dataListCell).toHaveClass('pf-v5-c-data-list__cell')
      : expect(dataListCell).toHaveClass(`pf-v5-c-data-list__cell ${testCase.class}`);
  });
});

test('Cell has text wrap modifiers when wrapModifier is passed', () => {
  [
    { wrapModifier: null as any, class: '' },
    { wrapModifier: 'breakWord', class: 'pf-m-break-word' },
    { wrapModifier: 'nowrap', class: 'pf-m-nowrap' },
    { wrapModifier: 'truncate', class: 'pf-m-truncate' }
  ].forEach((testCase, index) => {
    const testId = `cell-test-id-${index}`;

    render(
      <DataListCell data-testid={testId} wrapModifier={testCase.wrapModifier} key={index}>
        Primary Id
      </DataListCell>
    );

    const dataListCell = screen.getByTestId(testId);

    testCase.class === ''
      ? expect(dataListCell).toHaveClass('pf-v5-c-data-list__cell')
      : expect(dataListCell).toHaveClass(`pf-v5-c-data-list__cell ${testCase.class}`);
  });
});

test(`Cell has custom class when className is passed`, () => {
  render(
    <DataListCell data-testid="test" key={0} className="test-class">
      Primary Id
    </DataListCell>
  );
  expect(screen.getByTestId('test')).toHaveClass('test-class');
});

test(`Cell has ${styles.modifiers.alignRight} when alignRight is passed`, () => {
  render(
    <DataListCell data-testid="test" key={0} alignRight>
      Primary Id
    </DataListCell>
  );
  expect(screen.getByTestId('test')).toHaveClass(styles.modifiers.alignRight);
});

test(`Cell has ${styles.modifiers.icon} when isIcon is passed`, () => {
  render(
    <DataListCell data-testid="test" key={0} isIcon>
      Primary Id
    </DataListCell>
  );
  expect(screen.getByTestId('test')).toHaveClass(styles.modifiers.icon);
});

test(`Cell has ${styles.modifiers.noFill} when isFilled = false`, () => {
  render(
    <DataListCell data-testid="test" key={0} isFilled={false}>
      Primary Id
    </DataListCell>
  );
  expect(screen.getByTestId('test')).toHaveClass(styles.modifiers.noFill);
});

test('Toggle has aria label', () => {
  render(<DataListToggle aria-label="Toggle details for" id="ex-toggle2" />);

  expect(screen.getByRole('button')).not.toHaveAttribute('aria-labelledby');
  expect(screen.getByRole('button')).toHaveAttribute('aria-label', 'Toggle details for');
  expect(screen.getByRole('button')).toHaveAttribute('aria-expanded', 'false');
});

test('Toggle has aria-expanded set when isExpanded is passed', () => {
  render(<DataListToggle aria-label="Toggle details for" id="ex-toggle2" isExpanded />);
  expect(screen.getByRole('button')).toHaveAttribute('aria-expanded', 'true');
});

test('DataListAction renders to match snapshot', () => {
  const { asFragment } = render(
    <DataListAction aria-label="Actions" aria-labelledby="ex-action" id="ex-action">
      <Button id="delete-item-1">Delete</Button>
    </DataListAction>
  );
  expect(asFragment()).toMatchSnapshot();
});

test(`DataListAction has custom class when className is passed`, () => {
  render(
    <DataListAction aria-label="Actions" aria-labelledby="ex-action" id="ex-action" className="test-class">
      <Button id="delete-item-1">Delete</Button>
    </DataListAction>
  );
  expect(screen.getByRole('button').parentElement).toHaveClass('test-class');
});

test('DataListAction shows button with visibliity breakpoint set', () => {
  render(
    <DataListAction
      visibility={{ default: 'hidden', lg: 'visible' }}
      aria-labelledby="check-action-item2 check-action-action2"
      id="check-action-action2"
      aria-label="Actions"
    >
      <Button variant="primary">Primary</Button>
    </DataListAction>
  );

  expect(screen.getByRole('button').parentElement).toHaveClass('pf-m-hidden');
  expect(screen.getByRole('button').parentElement).toHaveClass('pf-m-visible-on-lg');
});

test('DataListAction hides button with visibility breakpoint set', () => {
  render(
    <DataListAction
      visibility={{ '2xl': 'hidden' }}
      aria-labelledby="check-action-item2 check-action-action2"
      id="check-action-action2"
      aria-label="Actions"
    >
      <Button variant="primary">Primary</Button>
    </DataListAction>
  );

  expect(screen.getByRole('button').parentElement).toHaveClass('pf-m-hidden-on-2xl');
});

test('DataListContent renders to match snapshot', () => {
  const { asFragment } = render(<DataListContent aria-label="Primary Content Details"> test</DataListContent>);
  expect(asFragment()).toMatchSnapshot();
});

test(`DataListContent has ${styles.modifiers.noPadding} when hasNoPadding is passed`, () => {
  render(
    <DataListContent aria-label="Primary Content Details" hidden hasNoPadding>
      test
    </DataListContent>
  );
  expect(screen.getByText('test')).toHaveClass(styles.modifiers.noPadding);
});

test(`DataListContent has hidden property when isHidden is passed`, () => {
  render(
    <DataListContent data-testid="test-id" aria-label="Primary Content Details" isHidden>
      test
    </DataListContent>
  );
  expect(screen.getByTestId('test-id')).toHaveAttribute('hidden');
});

test(`DataListContent has custom class when className is passed`, () => {
  render(
    <DataListContent data-testid="test-id" aria-label="Primary Content Details" className="test-class">
      test
    </DataListContent>
  );
  expect(screen.getByTestId('test-id')).toHaveClass('test-class');
});
