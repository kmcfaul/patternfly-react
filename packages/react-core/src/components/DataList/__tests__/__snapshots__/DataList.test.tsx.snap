// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell renders to match snapshot 1`] = `
<DocumentFragment>
  <div
    class="pf-v5-c-data-list__cell"
  >
    Secondary
  </div>
</DocumentFragment>
`;

exports[`Cells renders to match snapshot 1`] = `
<DocumentFragment>
  <div
    class="pf-v5-c-data-list__item-content"
  >
    <div
      class="pf-v5-c-data-list__cell data-list-custom"
      id="primary-item"
    >
      Primary Id
    </div>
    <div
      class="pf-v5-c-data-list__cell data-list-custom"
      id="primary-item"
    >
      Primary Id 2
    </div>
  </div>
</DocumentFragment>
`;

exports[`DataListAction renders to match snapshot 1`] = `
<DocumentFragment>
  <div
    class="pf-v5-c-data-list__item-action"
  >
    <button
      aria-disabled="false"
      class="pf-v5-c-button pf-m-primary"
      data-ouia-component-id="OUIA-Generated-Button-primary-1"
      data-ouia-component-type="PF5/Button"
      data-ouia-safe="true"
      id="delete-item-1"
      type="button"
    >
      Delete
    </button>
  </div>
</DocumentFragment>
`;

exports[`DataListContent renders to match snapshot 1`] = `
<DocumentFragment>
  <section
    aria-label="Primary Content Details"
    class="pf-v5-c-data-list__expandable-content"
    id=""
  >
    <div
      class="pf-v5-c-data-list__expandable-content-body"
    >
       test
    </div>
  </section>
</DocumentFragment>
`;

exports[`Item renders to match snapshot 1`] = `
<DocumentFragment>
  <li
    aria-labelledby="item-1"
    class="pf-v5-c-data-list__item"
    id=""
  />
</DocumentFragment>
`;

exports[`Item row renders to match snapshot 1`] = `
<DocumentFragment>
  <div
    class="pf-v5-c-data-list__item-row"
  />
</DocumentFragment>
`;

exports[`Renders to match snapshot 1`] = `
<DocumentFragment>
  <ul
    aria-label="list"
    class="pf-v5-c-data-list pf-m-grid-md"
    role="list"
  />
</DocumentFragment>
`;
